### ISSUE TITLE
The Future of Flashlight

### ISSUE THREAD

**Issue by nate-parrott** [nate-parrott/Flashlight#537]
TLDR: Hacking Spotlight is impossible in 10.11. [1] What now?

That's disappointing, but it's also a great opportunity to think about the future of Flashlight. 

Even when Flashlight didn't do that much, it was still useful because it was a *complement* to all of Spotlight's existing features. But now Flashlight's grown up — it has hundreds of features built by a great community. I think it's useful enough to survive as a standalone app — a fresher, more open-source take on all the [standalone](http://www.alfredapp.com) [launcher](http://qsapp.com) [apps](https://www.obdev.at/products/launchbar/index.html) we love. [2]

I still won't be able to work on Flashlight for a couple more weeks, so here's a challenge for all of you:

**Figure out what Flashlight can be once it's independent of Spotlight. And help me build it.** [3]

[1] Apple's new "rootless" security system blocks code injection into system processes. Technically you can disable rootless – but I'll have to agree with the [TotalFinder devs](http://blog.binaryage.com/el-capitan-update/) that it doesn't make sense to ask ordinary users to do this. And these security restrictions will only get tighter in the future – it doesn't make sense to build a product around circumventing them.

[2] Obviously we'll need to add things like app launching and file search, which Spotlight did for us. And we'll need to build a new UI. But that'll be fun.

[3] If you've got ideas, create issues for them and tag them with the *future* label. If you've got prototypes of standalone Flashlight UIs, add it in a new branch and submit a pull request with it.

**Comment by hufufuf** [comment]
More exactly, it seems that you need to disable the rootless security only to install SIMBL.
Once it is installed, you can enable rootless again.
And enable/disable SIMBL plugins as before.
See here : https://github.com/norio-nomura/EasySIMBL/issues/26#issuecomment-117028426

So it *could* be an acceptable think for power users.
But I understand it's not easy to ask each user to do something like this just to install a software. 

So, your proposition to built a new software from the ground with the best of spotlight and flashlight is great !
I just want something simple, clean (unlike Alfred, Launchbar & Cie) for which I can write easily some little plugins

**Comment by Sajonara** [comment]
As an "external" Software, there already is a similar solution: Quicksilver. It's source code is available at https://github.com/quicksilver/Quicksilver and it is extendable via plugins. I used Flashlight though, because it was very convenient not to have to use a separate program. Maybe one can think of extending Quicksilver or start a similar project from scratch.

**Comment by orbitly** [comment]
I've tried using both Quicksilver and Alfred, they're just not nearly as good as Flashlight injected into Spotlight was. I understand the need to appeal to average users, but disabling SIP in the recovery partition is not hard.

The new app should definitely use the same two column layout with immediate information as spotlight. Suggested websites would be nice, iTunes music library search would be great too. These are the things that stop me from wanting to use Alfred.

Thank you!

**Comment by brandonhorst** [comment]
You guys may want to look into Lacona. It is an extensible command runner with a powerful but simple NLP engine. Commands will be open source JavaScript, and it has its own UI.

Should be able to do everything Flashlight can do, but support more natural syntax. 

It's on Kickstarter now. And I wrote it, so sorry for the shameless self-plug. 

Link: https://www.kickstarter.com/projects/2102999333/lacona-natural-language-commands-for-your-mac
Command-building info: https://github.com/lacona/kickstarter-nerd-details

**Comment by yashendra2797** [comment]
I love Nate's work, but I have to agree with the community. I loved Flashlight since it was integrated with Spotlight. While I would love to see what Nate has planned, I'm afraid I may not use flashlight if it is made into a standalone app. There are many others which do the same work. The beauty of Flashlight was in its simplicity.

**Comment by evrial** [comment]
Have you asked Apple about opening Spotlight API to external plugins? Why they prohibit such idea?

**Comment by xremix** [comment]
@nate-parrott thanks for your kind words and sharing your opinion on the future of Flashlight.

If it's the case, that only the installation of SIMBL requires to disable rootless, I'd love to have a solution where the plugins could work with both, A solution that integrates with spotlight (for power user) as well as a separate tool with an own UI.
They could both use the same plugins as well as the plugin manager.
Any thoughts on this?

**Comment by sohocoke** [comment]
As a launcher addict since the days when LauchBar was the only game in town, and having extensively used Quicksilver, Alfred over many years, I have to agree with the opinions already cited that having another launcher app is a big downside.

Would it be at all possible to re-implement the current hop-on-spotlight behaviour by using the Accessibility API to get updates on what the user's entering, then neatly clobber Spotlight output with Flashlight output? It would be a cool hack if it can work.

**Comment by bmehling** [comment]
Generally agreeing w/ other posters. I was a long-time quicksilver, then alfred (and now again) user. I really liked having a single search/launcher -- Flashlight's integration into Spotlight is a pretty killer feature.

Perhaps @xremix 's idea of universal plugin support between a standalone launcher and a version we have to hack into spotlight is the way to go?

**Comment by xremix** [comment]
I'd love to see such a solution. But it would require more effort of the community so we would need more people that commit with us.

**Comment by Bisewski** [comment]
I agree with was said: Flashlight is awesome because it's simple and is almost invisible. It just works, like a OS X feature. There's nothing like CMD+Space and voilà! Since I setup Flashlight, I changed (for very best) the way I do almost my daily tasks in front of my Mac. And don't have it anymore, is very sad :(

I think @xremix is right. I guess there is a lot of power user who love Flashlight, and run some Terminal line is not a problem for most of them. These can keep using Flashlight via Spotlight. The others, would have a standalone app.

I'm here for what you need. Count with me.

**Comment by wilmoore** [comment]
Honestly, this is really a tool that should be built-in. Flashlight as it used to work is so seamless. I am a paid user of Alfred and I have to say, once I tried Flashlight, I completely uninstalled Alfred (no offense to the Alfred developers -- it's great at what it does, but the approach is similar to other products which is not as seamless as Flashlight).

I think that if Flashlight goes in the direction of the other launchers, there is little (besides a good ecosystem of plugins and that it is open-source) to differentiate. Sure, free is nice, but Let's be honest; for OSX users, free is generally not a deal-breaker.

Perhaps if it goes to a full standalone app, I'd still stick with Flashlight, but then again, I've paid for Alfred so perhaps I just stick with that. I'm not 100% sure, but keep in mind, when Flashlight was working, I was 100% sure that I'd be sticking with Flashlight and I was regretting that I had paid for Alfred.

I hope this opinion helps your decision somehow @nate-parrott.

**Comment by Lanzafame** [comment]
@nate-parrott I don't know if you have already tried or whether it is even possible but have you approached Apple about an 'official' work around? I agree with the above stated responses, that the integration into Spotlight is what made Flashlight better of all the options. But yeah just a suggestion.

**Comment by acaloiaro** [comment]
I'm neither a Flashlight user (I was hoping to be, but alas, The Captain said no), nor have I developed against the Core Spotlight Framework—so bear with my naïvety. However, Apple's [Core Spotlight Framework](https://developer.apple.com/library/prerelease/ios/documentation/CoreSpotlight/Reference/CoreSpotlight_Framework/) seems to provide the necessary hooks. The implementation may be convoluted in that a standalone Flashlight app would create files to be indexed by Spotlight, and NSUserActivity would allow one to navigate from Spotlight into specific Flashlight states/navigation points. Is this workflow too convoluted? By all means, punch some holes in this suggestion. Is it worth considering? Cheers.

**Comment by xdissent** [comment]
@acaloiaro that's iOS though, right? AFAIK there's no OS X equivalent.

**Comment by acaloiaro** [comment]
@xdissent You're right, I hadn't noticed the URL. However, I think the general approach still applies: 

1) [Create Metadata Attributes](https://developer.apple.com/library/prerelease/mac/documentation/Carbon/Reference/MetadataAttributesRef/MetadataAttrRef.html#//apple_ref/doc/uid/TP40001689)
2) [Create a Flashlight Importer](https://developer.apple.com/library/prerelease/mac/documentation/Carbon/Conceptual/MDImporters/MDImporters.html#//apple_ref/doc/uid/TP40001267)
3) [Provide a Quick Look for Flashlight files](https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/Quicklook_Programming_Guide/Introduction/Introduction.html#//apple_ref/doc/uid/TP40005020)

The next step would be to determine whether or not one can extract additional information from the query when displaying Quick Look result in Spotlight. So, for example, when one types "weather in New York", a Quick Look showing the Weather file populated with weather data for New York is displayed in Spotlight.

**Comment by tobsn** [comment]
So why did Lacona fail its kickstarter funding and why did nate not support it?

**Comment by brandonhorst** [comment]
Lacona dev here.  I did reach out to Nate (via email) to consider [Lacona](http://lacona.io/) as a possible successor, and never got a response. But, of course, that's perfectly fine.Lacona is a different project with different goals.

Lacona failed it's kickstarter but [development is going to continue](https://www.kickstarter.com/projects/2102999333/lacona-natural-language-commands-for-your-mac/posts/1336040), and I believe there is a place for both Lacona and Flashlight in the future.

**Comment by tobsn** [comment]
@brandonhorst if you set up a donation button... that might be a good idea :)

**Comment by Nxt3** [comment]
I would definitely like to see where Lacona and Flashlight goes. The two would be great together. Lacona already has a fantastic UI concept; it's definitely something to build off of.

**Comment by brandonhorst** [comment]
@Nxt3 @tobsn thanks, guys! I appreciate the kind words. I did add donation buttons to the website, but I'm not going to push for donations much until I actually release something - I want to make sure I keep everything on the up and up.

This isn't the place to talk about this, though. If you have questions, you can bring them here: https://github.com/lacona/www.lacona.io/issues/3

**Comment by suokun** [comment]
Why not join Apple spotlight team. Your work is so great and you can make spotlight much stronger based on flashlight.

**Comment by TP18** [comment]
He has joined Apple

**Comment by suokun** [comment]
Cool, maybe we can expect more functionalities as Flashlight in Spotlight in the future.

**Comment by orbitly** [comment]
He was at Apple for a summer internship, that's why he couldn't work on hacky software over the summer. Should be over by now.

**Comment by MrLoh** [comment]
It seems like Total Spaces has found a way to [disable System Integrity Protection](http://totalspaces.binaryage.com/elcapitan), so they can still alter the Dock app. Not sure wether Flashlight has the same issue and could have a similar fix, and even if so it would be very dirty, but maybe it's an inspiration to someone.

**Comment by Nxt3** [comment]
@MrLoh We already know how to disable System Integrity. It's not hidden; it's a known way of disabling it. Nate is saying that it isn't worth asking everyone to disable SIP to use software.

**Comment by tsimmons** [comment]
Apparently [Bartender 2](http://www.macbartender.com/system-item-setup/) has found a way around it without requiring the end-user to do anything special (perhaps the installer does the SIP magic behind the scenes?) Has anyone looked into what Surtees Studios is doing to accomplish their El Capitan compatibility?

**Comment by Nxt3** [comment]
@tsimmons Bartender had Spotlight integration? I'm interested in how they accomplished being able to control the system icons without disabling SIP too.

**Comment by sadokx** [comment]
Just found out about Flashlight, and was disappointed it no longer works on my mac. Having a separate app is useless, I'm sorry. I think the answer is simple: Transition Flashlight into a power-user app. That way we can disable the security issue and move on with our lives. I guess Flashlight is mostly used by power users anyway. 

PS Is there a way to make Flashlight work currently? How to disable the security issues?

**Comment by orbitly** [comment]
I think most of the people here would be fine with disabling SIP to use Flashlight. We haven't had SIP for the past 15 years of OS X, why is it suddenly essential?

**Comment by RF18** [comment]
Firstly, thanks Nate for this amazing app you've developed. Honestly it has changed my productivity at work immensely-hard not to go a day without relying on this. Kinda bummed apple doesn't realize how ingenious this little tweak is and that they haven't taken any steps in producing similar results but again i'm not expecting much from their side. 

So i tried some of the solutions above - I tried disabling csrutil (debug only at first and than the whole thing) but the symbl would still not inject into the app (it was saying the spotlight version/ flashlight app is not compatible). I read that someone tried changing the source code to add the new spotlight version but it didn't help.

 Anybody else find a way to get flashlight running up in 10.11 or i guess i'll just revert back to 10.10.5 or whatever the yosemite latest was. (El capitan really doesn't seem to be worth the update so far)

Thanks guys for any inputs

**Comment by MrLoh** [comment]
Bartender also asks you to turn off SIP and if you want to change any Apple App Icons you will also have to turn it off. Seing how many things don't work without SIP, I'm pretty sure most powerusers will turn off SIP. I will, I've lived without it for the past and will survive without it for now. Obviously there are other things that need to be adapted for Flashlight to work under El Capitan, but I don't think that not many people will turn off SIP, no everage Joe is using Flashlight anyways.

**Comment by MrLoh** [comment]
Sorry, you're right Bartender 2 has found another way. That sounds promising. I'd really like to get back to using Flashlight. But it seems the main issue is figuring out what else changed. Sadly I have no experience with Mac App Development, but I'd be glad to help in any way, if someone takse the lead.

**Comment by norlin** [comment]
AFAIK, Installer can modify system files (when installing Apple-signed packages), no? Have you tried to contact them to move Flashlight into official OSX App?

(There was already similar questions above, but no answer from @nate-parrott... )

**Comment by qinyuhang** [comment]
hey！
I think make it official app is a great idea！

发自我的 iPhone

**Comment by wdanilo** [comment]
I was using Flashlight also just BECAUSE it was integrated into Spotlight. I do not will use yet another "launcher - like" app, when spotlight is developed in the right direction (natural language recognition etc). Flashlight was amazing because of its integration. I think Apple will do something with the new system integrity protection, because a lot of apps suffer form it, including really good ones like totalspaces, total finder or flashlight. Anyway, going into yet another standalone app is just wrong.

**Comment by norlin** [comment]
@wdanilo That's for sure! I just mentioned it will be good to have the same (I'm talking about functionality/integration/etc) Flashlight packed into "official App" package with install via App Store.

**Comment by hgpt** [comment]
I'm so sad that the project turned out to be abandoned with the introduction of OS X El Capitan. I really don't care if I'll have to disable SIP to make it work. I survived 15 years without SIP and had no problem at all. From my point of view Flashlight was used mostly by power users anyway, so abandoning the project because of regular users not being able to deal with disabling SIP it's not a point. 
When it comes to Flashlight being developed as a standalone app, that's another bad idea if you ask me. The beauty of Flashlight, apart from the many plugins, was Spotlight integration! Many of us are loving this app (or loved it) for the Spotlight integration. Standalone app may prove to be a mistake.
Now, has any of you managed to make Flashlight work with OS X El Capitan? I miss my trusty Flashlight so so much. :(
P.S: Thanks Nate for all the work that you did! We really appreciate it! Where is that donate button?

**Comment by brechtm** [comment]
Personally I also see little value in a stand-alone app. An official plugin interface in Spotlight by Apple would be ideal, but I'm not sure if this could ever materialize.

If you're interested in getting Flashlight working on El Capitan, please see issue #545.

**Comment by mesqueeb** [comment]
We don't want a standalone app. You've got alfred for that. We don't mind turning the SIP off, can someone please relaunch this with similar instructions to Bartender 1?

**Comment by evindor** [comment]
Just discovered Flashlight today, it's brilliant! Please please please, reanimate this project for El Capitan, the hassle with disabling SIP for installation is nothing compared to how much time Flashlight can save you on a daily basis.

**Comment by xremix** [comment]
Really interesting that some many are interested in a Flashlight that works with the native Spotlight and don't hesitate to disable SIP. I'd love to see Flashlight running again, as it improved my daily work and I spent a lot of time in building plugins.
So far I can't imagine that Apple would open the Spotlight for that sort of plugins, but please disabuse myself :)
What I really would be interested is if it's really the average Flashlight user who is speaking here, or if most of the people talking here are the power users because they are using github and contribute with their opinion, or if most normal users just don't manifest their opinion here (just look at the weather plugin, there seem to be more than 74k users).

Also I'd love to Nates current opinion after all these responses came in.

I personally feel more and more that a Spotlight extension is the best way to go, instead of building a new clone of all these products that are already out there.

**Comment by wdanilo** [comment]
I can only vote for 4 other people I know, that do NOT use github and red to use Flashlight - they've already doable SIP to use TotalFinder and TotalSpaces, being "average" users. To be honest with you, SIP in its current form is just wrong and I'm sure it will be changed in the future. Additional Flashlight would be a very good point for Apple to change SIP policy. And one more thing - if somebody is using Flashlight, he is a some-kind of power-user and such power users is more than you can think of! :D

**Comment by jdahdah** [comment]
My 2ct, for what it's worth: you're basically taking the only thing that made Flashlight unique and throwing it out of the window. We don't need another Quicksilver or Alfred or Launchbar or Lacona. We need a more powerful, real Spotlight, and that's the whole reason for Flashlight's existence. I understand that you have looked into all of the technical possibilities and that this situation is Apple's fault, but rather than building yet-another-launcher-app (as great as it might be), I would rather see all of this valuable energy go into lobbying Apple to build a decent, plugin system into Spotlight itself. I like their lightweight, privacy-friendly approach to Content Blockers and believe that they are now more open than ever to these kinds of extensions.

**Comment by norlin** [comment]
My suggestion was not to make it "standalone" app, rather just to "pack" existing behaviour into official app package, with using official Installer features to extend Spotlight. (i'm still not sure it's possible, but i have seen Installer could bypass SIP).

**Comment by pencilcheck** [comment]
It's sad to see spotlight coming to an end. Standalone app loses the beauty of simplicity. But why can't spotlight just be another Alfred? Alfred makes it very easy to rebind the key binding to start working like apple spotlight. If spotlight is making a standalone app, a UI that resembles the original spotlight is a must, then integration comes second, plugins third.

**Comment by brechtm** [comment]
Perhaps a good compromise would be to create a stand-alone Flashlight that hands control over to Spotlight if none of the enabled plugins are triggered.

Suppose I have the *Remind Me* and *Google Translate* plugins installed. The stand-alone Flashlight should be configured to pop up on Spotlight's usual hotkey, say Cmd+Space. Spotlight's hotkey needs to be changed to something else, say Ctrl+Cmd+Space. If I now press Cmd+Space, Flashlight's window pops up and I can enter some commands:
- `remind me to go shopping` will activate the *Remind Me* plugin, or
- `gtr ennl shoelaces` activates the *Google Translate* plugin, or
- entering anything not starting with `remind me ` or `gtr ` closes the Flashlight window even **before pressing enter**, open Spotlight and paste the entered text into Spotlight's text input. All following keypresses are simply received by Spotlight.

Feedback welcome... I'm a little worried that some keypresses might get lost when handing over control to Spotlight. Except for that, this should be fairly simple to implement.

I would still prefer to have Flashlight fully integrated in Spotlight, provided this is possible by installing Flashlight with SIP disabled (and enabling it again after installation), but the suggested approach would be more robust against future moves by Apple.

**Comment by alpha-ninja** [comment]
how about this: opening flashlight triggers spotlight, but flashlight is displayed over the spotlight text field. Then, as the user types, flashlight might display its own suggestions/actions.
If none exist, the text field from flashlight stays, but spotlight receives the text typed into flashlight.

**Comment by orbitly** [comment]
As it seems Nate has abandoned this project, to any software developers, I'm sure plenty of people here would pay good money for a working fork on 10.11, even if we have to disable SIP.

**Comment by inket** [comment]
Here's a thing though: You don't have to disable SIP at all.

**Comment by orbitly** [comment]
@inket Do you know how to get it working?

**Comment by inket** [comment]
Get Flashlight working? No.

You'd need to rewrite it because it's based on EasySIMBL which doesn't work anymore. It needs to use SIMBL instead. Should work in theory.

For how to install it, just make it into a .pkg, boot into Recovery, install the .pkg through Terminal and restart. You technically don't need to disable SIP.

**Comment by WildElf** [comment]
SIMBL /also/ only works by disabling SIP. 
http://evotec.pl/mac-os-x-el-capitan-how-to-get-simbl-working-after-upgrade/

I like a good theory, but theories also require evidence. Do you want to provide some evidence of your genius, @inket?

**Comment by inket** [comment]
@WildElf If I provide evidence then, would that make me a genius? You don't have to be passive-aggressive.

Here's Safari 9 with 2 SIMBL plug-ins I made: one to limit the tabs' width (you can see the source code for that on my profile), and one I'm working on to make pinned tabs customizable.

Safari is supposed to be protected against code injection by SIP.

<img width="658" alt="screen shot 2015-11-02 at 12 13 11am" src="https://cloud.githubusercontent.com/assets/679224/10869563/8eeb450a-80f6-11e5-8a9a-d507ad1b7a2c.png">

**Comment by avaidyam** [comment]
@inket Random Q, how'd you style your bash + Terminal like that?

**Comment by tsimmons** [comment]
@inket Also, have you gotten Flashlight to work using your method of installation for SIMBL and this method https://gist.github.com/selaromi/59666a71387d39b5bfac of running Flashlight?

**Comment by zhenyulin** [comment]
Would like to know how to get the theme as well, if it is available.

**Comment by WildElf** [comment]
And a genius was born, @inket! Sorry to be the bearer of bad news, but that's how it works, historically and by definition. 

If this just seems more passive-aggression, then the additional paragraphs of this post will just be more aggravating semantics you need not waste your time with. Thanks for pointing me to your code! I'll look for the evidence I seek there.

Aristotle, Da Vinci, Newton, Einstein, every scientific genius in history was labeled as such because of the evidence they provided, and many authors, poets, and other creatives only for the work they produced.

Genius, from it's latin roots means "to bring into being, create, produce." Although it has since come to mean innate talents of an individual, no talent has been recognized by restricting oneself to hypotheticals, no matter our claims to talent.

Being a romantic, I set a very low bar for genius, myself, so I'm surprised that the compliment causes you offense. But since it does, I will call you genius no more. Leave it to the history writers to decide, and let us trade only in evidence instead of name-calling.

I apologize for being such an aggressive nuisance as to ask if your actions can back up your words (even though many great works are only recognized because of some nuisance who pressured the creator to publish, e.g. Newton, being pressured into greatness seems to cause you stress which you take out on convenient targets).

Although, it seems others clamor for even more evidence that your claims are valid, I won't presume to assault you further, or even make myself a target, by agreeing with their requests.

**Comment by paulrudy** [comment]
I've subscribed to this thread to keep up on the status of Flashlight... hope that it can manage to stay focused on that.

**Comment by Nxt3** [comment]
@avaidyam He used an emoji for the arrow; everything else is just customization done with a .bash_profile.

**Comment by inket** [comment]
> @inket Also, have you gotten Flashlight to work using your method of installation for SIMBL and this method https://gist.github.com/selaromi/59666a71387d39b5bfac of running Flashlight?

Well, yeah. It does inject properly but it doesn't work with the new Spotlight because it's incompatible.

<img width="768" alt="screen shot 2015-11-07 at 11 02 28am" src="https://cloud.githubusercontent.com/assets/679224/11012900/364bbc3e-853f-11e5-8f03-73c2142d31e8.png">

But it's no surprise because the end result of both methods of installing SIMBL is the same. The only reason the initial method disables SIP is to be able to write to system folders, which you can already do from another OS (like Recovery) *without* disabling SIP.

Honestly we should be way over this step by now, and looking into ways to make the SIMBL installation unattended.

@ericjzim You got issues.
@Nxt3 [inket/terminal-setup](https://github.com/inket/terminal-setup)

Let's keep the discussion to Flashlight, thanks.

**Comment by avaidyam** [comment]
@inket You're totally right, I will take a look at this and maybe fork it and see what happens.

**Comment by orbitly** [comment]
https://github.com/nate-parrott/Flashlight/tree/menubar

This is a fork nate worked on a month ago in attempt to get it working. You can see he removed the check Spotlight version, and tons of other changes. I can't get it to build in Xcode though

**Comment by DuPupu** [comment]
I think a UI and a set of symbology like the auto-complete command input line in Bloomberg Terminal can be much more convenient for hackers. 

It use 2-4 chars acronym to represent a function, space for split, and remaining parameters followed. 

Such as 

**GP MMM US Equity** for the graph of 3M stock
**HP MMM US Equity** for the history price table of 3M stock
**DINE** for restaurant review, launch a window functioning similar with Yelp

The operation is really efficient, and it is cool

**Comment by gianvecchia** [comment]
If I disable rootless, can I make Flashlight work on my OS X 10.11?

**Comment by hgpt** [comment]
Any news? Is there any hope left for this amazing tool?

**Comment by zarkone** [comment]
![So sad](http://www.comicsbeat.com/wp-content/uploads/2015/07/tumblr_inline_nc66zff7SU1r9c2d4.gif)

**Comment by brandonhorst** [comment]
@hgpt @zarkone I apologize for the shameless self-promotion, but [Lacona](http://www.lacona.io/) launches a week from today. It may fill some of your needs post-Flashlight.

**Comment by eirnym** [comment]
@brandonhorst will be it integrated into Spotlight?

**Comment by brandonhorst** [comment]
@eirnym Nope. It's an independent tool. I'm afraid any Spotlight integrations will be go the same way as Flashlight.

**Comment by avaidyam** [comment]
@brandonhorst I'm sorry I'm saying it like this, but that may mean DOA to a lot of people.

**Comment by brechtm** [comment]
@brandonhorst Please consider fallback to Spotlight as described in [this comment](https://github.com/nate-parrott/Flashlight/issues/537#issuecomment-149228336).

**Comment by tsimmons** [comment]
@brandonhorst Will your app work like Alfred 2 or other launcher apps?

**Comment by glitchninja2000** [comment]
@brandonhorst @brechtm I think [brechtm's idea](https://github.com/nate-parrott/Flashlight/issues/537#issuecomment-149228336) is probably the best one. I've been looking into integrating Flashlight into an older of Spotlight. If anyone has an update where Flashlight works, please send my the Spotlight app located in HD/System/Library/CoreServices/Spotlight.app. I've been able to run two instances of Spotlight already, so that wouldn't be a problem. There is the issue of quiting the second Spotlight, since you have to use `killall Spotlight`, but I'm hoping using an old native Spotlight would be able to work.

**Comment by brandonhorst** [comment]
@avaidyam I understand. I think that it's a necessary evil at some point. In my own work, I've found that Alfred (and now, my yet-unreleased build of Lacona) covers all of my spotlight use-cases by duplicating the most common functionality. But I understand that this won't be true for many people.

@brechtm @glitchninja2000  That is a good suggestion and may come at some point in the future, but the Lacona interface is pretty distinct from spotlight - focusing on Natural Language parsing rather than keywords. But it may come at some time version 2+.

@tsimmons Yeah, it works similarly to Alfred, just much more powerful. The beta launches on Friday (the 29th), so you can try it out then!

**Comment by The-Penultimate-Defenestrator** [comment]
With iOS 9, apple introduced a whole big thing for developers to integrate their apps with spotlight, right? I think there's a good chance they're planning to put something similar in OS X 10.12. Anyway, we'll see in June.

**Comment by vlux** [comment]
Hope the FlashLight Long Live!
@brandonhorst  I came from the Alfred. Compared these two, the FlashLight is more suitable for my work. So sad that I came late and the 10.11 destroy everything. Hope see ur guys future awesome works. :+1:

**Comment by mesqueeb** [comment]
The difference with Alfred was that flashlight shows information directly in the search box. e.g. google image search would show images straight in the spotlight search box.

Is there anyone who can make an app that does this that is just as fast as flashlight was?

**Comment by brandonhorst** [comment]
@mesqueeb http://lacona.io/ is getting there! Not there yet, unfortunately. But it's improving every day!

**Comment by Kiura** [comment]
hey, great app, loved flashlight, and lacona seems to be a good replacement, but the thing I miss the most from flashlight is being able to see the results within the app and some plugins. Are the feature and plugins will be added to lacona in near future?

**Comment by Kiura** [comment]
Thx for info, great job! )

**Comment by jcampbell05** [comment]
Is it possible to override the spotlight shortcut ?

**Comment by revolter** [comment]
@jcampbell05: Uncheck `System Preferences > Keyboard > Shortcuts (top bar) > Spotlight (left bar) > Show Spotlight search` and then I suppose you can set <kbd>cmd</kbd>+<kbd>space</kbd> for `Lacona`, but I can't test it right now.

**Comment by Shawzborne** [comment]
@brandonhorst Is there a way to get Lacona to search network drives? Such as a local Synology NAS?

**Comment by nabn** [comment]
@revolter's method works.

**Comment by brandonhorst** [comment]
@Shawzborne Lacona uses the normal Spotlight index - it should find the same files/apps/whatever that Spotlight finds.

I don't want to have too much Lacona discussion on this thread. Could we move discussion to [laconalabs/LaconaApp](https://github.com/laconalabs/LaconaApp/issues)?

**Comment by JMY1000** [comment]
@brandonhorst Seconded.

**Comment by w0lfschild** [comment]
Here's a fork for 10.10+ 

Try my fork https://github.com/w0lfschild/Flashlight

**Comment by JMY1000** [comment]
@w0lfschild Does it work on 10.11?

**Comment by controversial** [comment]
@JMY1000 That's what 
> 10.10**+**
would imply

**Comment by JMY1000** [comment]
@controversial Usually, but given all the trouble people have been having getting it to work on 10.11, I just wanted to make sure.

**Comment by controversial** [comment]
Ok. I haven't actually tried it myself, but I assume it does

**Comment by matthias-vogt** [comment]
Totally works on el capitan, super cool

**Comment by avaidyam** [comment]
But does it work on macOS 12 Sierra? 🙄

**Comment by tsimmons** [comment]
What were your steps to get it running on El Cap? Did you install SIMBL first, etc.?

**Comment by orbitly** [comment]
Thank you so fucking much, w0lfschild!

**Comment by w0lfschild** [comment]
@tsimmons There are install instructions in the repos README. 

Here are some even more clear instructions.
1. Download [mySIMBL](https://github.com/w0lfschild/app_updates/raw/master/mySIMBL/mySIMBL_master.zip)
   - Unzip
   - **_Disable System Integrity Protection_** (Only if using 10.11+)
   - Open `mySIMBL.app`
   - Enter password to install SIMBL
   - **_Enable System Integrity Protection_** (Only if using 10.11+)
2. Download [Flashlight](https://github.com/w0lfschild/Flashlight/raw/master/build/Flashlight.zip)
   - Unzip
   - Open  `Flashlight.app`
   - Install the plugins you want to use
3. Open Spotlight and start using _Flashlight_ 

![screen shot 2016-06-13 at 1 19 37 pm](https://cloud.githubusercontent.com/assets/1920148/16021819/8eda46d2-3169-11e6-9a1f-5384a7f6ccb3.png)

**Comment by tsimmons** [comment]
@w0lfschild Perfect! This is awesome.

**Comment by Bisewski** [comment]
Awesome! You saved lives :D

**Comment by wickles** [comment]
This is better than WWDC!

**Comment by JMY1000** [comment]
@w0lfschild You are amazing. Thanks so much.

**Comment by xremix** [comment]
Awesome! I guess this is the best news of the day and I am so glad to get this working again. @w0lfschild, thanks a lot for your effort and getting this back to live for me and all the others.

**Comment by RF18** [comment]
@w0lfschild Can't even express enough thanks! Great breakthrough! Confirmed working on 10.11.5

PS, quick question for you guys- I vaguely remember flashlight being able to navigate google searches and than clicking links within spotlight and never having to open a web browser-but right now when you navigate through flashlight and click on a link from a search- it opens up on a new window in your browser instead of through the spotlight interface. Has something been lost in transition to get a working flashlight for 10.11.x? anyways it's not a major issue just wondering if maybe my plug in needs to be reinstalled or it doesn't work for you guys as well. 

Thanks again though @w0lfschild!! Back to power using!

**Comment by w0lfschild** [comment]
@avaidyam 

> But does it work on macOS 12 Sierra? 🙄

As of the first build yes:

![screen shot 2016-06-13 at 4 23 52 pm](https://cloud.githubusercontent.com/assets/1920148/16026460/abca3210-3183-11e6-81a2-8e4a68dc0639.png)

**Comment by remipeyronmusso** [comment]
Awesome news ! Thank you very much, works perfectly !

**Comment by foomep** [comment]
I need to Disable System Integrity Protection (link https://github.com/nate-parrott/Flashlight/issues/537#issuecomment-225696255) on Sierra?

**Comment by w0lfschild** [comment]
@foomep yup all systems above Yosemite come with System Integrity Protection which needs to be disabled once to install a helper.

**Comment by rafalenden** [comment]
For some reason after following instruction my icons in Chrome changed:
![image](https://cloud.githubusercontent.com/assets/4199239/17498382/edaa30ba-5dbe-11e6-8e95-7543277b1f37.png)

**Comment by wickles** [comment]
Not sure what that has to do with this? But that's just the new material look in Chrome, made default in v51/52+ depending on your OS. You can disable it in chrome:flags...

**Comment by rafalenden** [comment]
@wickles thank you, my flag was set to "Default" so I had to change it to "non-material", I was just confused, it happened after disabling the System Integrity Protection.

**Comment by LouisCAD** [comment]
@rafalenden Beacause the Chrome update was installed, and changes were applied after your reboot restarted Chrome too

**Comment by vhristev** [comment]
I tried on Sierra but its not working for me: 
OS X version: 10.12 (16A323)

Disable SIP 
Install mySIMBL and i got some error at the first install but after the 2nd it fine. 
Now in mySIMBL i see plugin SpotightSIMBL is enabled.
Start Flashlight and tried some of the plugins but its not working. 

I enable SIP again but at some reason its not working for me. Back in Yosemite days when I was using Flashlight i notice regular crashes and after disable / enable its back again but this workaround its not working.

**Comment by vhristev** [comment]
I uninstall everything and start from scratch and now Flashlight is working don't know what was the root cause.

**Comment by mesqueeb** [comment]
I hope Flashlight will ever come out properly again. I can't wait!!
It was the best thing since bread came sliced. My favourite function was: `gi anything` e.g.: `gi suez canal` for INSTANT image results without having to open a browser.

What was yours?

**Comment by JMY1000** [comment]
@mesqueeb I doubt we'll see an official release by nateparrot. Check out [w0lfschild's fork.](https://github.com/w0lfschild/Flashlight)
